src/main/java/org/folio/kafka/KafkaHeaderUtils.java

```diff
@@ -42,4 +47,15 @@ private KafkaHeaderUtils() {
           (a, b) -> StringUtils.isNotBlank(a) ? a : b)));
   }
 
+  /**
+   * Retrieve distinct value by key
+   *
+   * @param keyExtractor function to get a key
+   * @return returns a predicate that maintains state about what it's seen previously, and that returns whether the given element was seen for the first time
+   */
+  private static <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor) {
+    Set<Object> seen = ConcurrentHashMap.newKeySet();
```

Comment:
Why we need to use ConcurrentHashMap here, can we use just common Map?

Reply:
In general, stream filter method is not thread-safe and there is a possibility of encountering a problem when headers may disappear from returned list.
it doesn't have much performance impact at all.

Reply:
I agree that we would not see any difference for performance. If we have this map as shared variable, means declared as field - totally agree that we need to use concurrent collection.
In our case its local variable that is stored in each thread's own stack. That means that local variables are never shared between threads. That also means that all local primitive variables are thread safe.

Reply:
Made changes as discussed. 
Thank you for being attentive.